client/relay-chain-minimal-node/src/blockchain_rpc_client.rs

```diff
@@ -317,9 +325,9 @@ impl RuntimeApiSubsystemClient for BlockChainRpcClient {
 		at: Hash,
 	) -> Result<
 		Vec<(
-			polkadot_primitives::v2::SessionIndex,
-			polkadot_primitives::v2::CandidateHash,
-			polkadot_primitives::v2::DisputeState<polkadot_primitives::v2::BlockNumber>,
+			polkadot_primitives::v3::SessionIndex,
+			polkadot_primitives::v3::CandidateHash,
+			polkadot_primitives::v3::DisputeState<polkadot_primitives::v2::BlockNumber>,
```

Comment:
Did you intend to leave this?

```suggestion
			polkadot_primitives::v3::DisputeState<polkadot_primitives::v3::BlockNumber>,
```

Reply:
Wow, nice catch! How does it even compile in this form? :man_shrugging:

Reply:
I've found several more and fixed them. I hope it will become less messy after paritytech/polkadot#6480.

Reply:
DQ but could you mark the `V2` as deprecated and rely on the linter to find all of them?

Reply:
Well it is not *totally* deprecated, we still have perfectly legal `v2::SessionInfo` and `v2::OldV1SessionInfo`... But I wonder how it was even compiling, all the primitives are moved to `v3` and are not reexported from `v2`... Need to check my build environment consistency, something went awry.

Reply:
Ah, I see. My bad, I was `pub use`ing some primitives in `v2` on the Polkadot side instead of just `use`ing them. I'll fix it and check against Cumulus build again.